packages/embark-coverage/src/lib/printer.ts

```diff
@@ -0,0 +1,29 @@
+import fs from "fs";
+
+import parser from "solidity-parser-antlr";
+import { printers } from "prettier-plugin-solidity";
+import prettier from "prettier";
+
+export class Printer {
+  constructor(private ast: parser.ASTNode) {}
+
+  public print() {
+    const { ast } = this;
```

Comment:
I assume we're doing this due to scoping issues when using `this.ast` instead below?

Reply:
That's correct. This reads nicer than `const self = this`, as we really only need that one property.